### ISSUE TITLE
Don't automatically enable pattern synonyms extension

### ISSUE THREAD

**Issue by vmchale** [ndmitchell/hlint#474]
Hi,

As I understand it, `hlint` parses using `haskell-src-exts` and enables many extensions by default. Unfortunately, this can cause otherwise valid Haskell to fail to parse. As an example, consider the following:

```haskell
module Pattern where

pattern :: String -> String
pattern = (++ "x")
```

This module compiles fine with `ghc`. Unfortunately, it triggers a parse error in `hlint`, viz.

```
Pattern.hs:3:9: Error: Parse error
Found:
    module Pattern where
  > pattern :: String -> String
    pattern = (++ "x")
1 hint
```

If, on the other hand, I run `hlint -XNoPatternSynonyms` everything works fine.

I'm not sure what the ideal solution would be here, but I think it might be good to disable pattern synonyms by default since it breaks valid Haskell2010.

Thanks!

**Comment by ndmitchell** [comment]
Lots of extensions break a little bit of Haskell2010 in the edges - it's a balancing act figuring out whether you would get more stuff broken if you turn it on or off. For pattern synonyms I think it's probably a good idea to rename your code to avoid pattern, since a future Haskell might break it. It only breaks a small number of modules, so the overall feeling is that it should be on by default.

You can always pass it as a command line flag, or add it to `.hlint.yaml`, or even pass `-XHaskell2010` if you think that is the right default for you.
